k8s-dev/cstor-pool-mgmt/deployment-cstor-pool.yaml

```diff
@@ -0,0 +1,72 @@
+# The deployment for cstor-pool and cstor-pool-mgmt.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cstor-pool
+  labels:
+    app: cstor-pool
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cstor-pool
+  template:
+    metadata:
+      labels:
+        app: cstor-pool
+    spec:
+      containers:
+      - name: cstor-pool
+#The image name here will be changed to openebs once the business logic pr is reviewed and merged in openebs/maya.
+        image: gkganeshr/cstor-pool:v1.1
+        ports:
+        - containerPort: 80
+        securityContext: 
+          privileged: true
+
+        volumeMounts:
+        - name: device
+          mountPath: /dev     
+        - name: tmp
+          mountPath: /tmp
+#To avoid clash between terminating and restarting pod, we keep initial delay.
+        lifecycle:
+          postStart:
+            exec:
+              command: ["/bin/sh", "-c", "sleep 15"]
```

Comment:
as I understand it, this sleep is needed only for the sidecar. pool container does not import any pool when it is started without the sidecar telling it to do so. However I'm not sure if my understanding is correct.

Reply:
@jkryl Is it okay that two zrepl running and pointing to same pool for particular duration while pod restart?

Reply:
it is not ok if they both have the pool imported. To prevent this, the timeout in sidecar container should be sufficient. IIRC we agreed that zrepl will not import any pools when started (because there is no cache file in /etc/zfs). Once sidecar is up and running it will try to import a pool.